### ISSUE TITLE
WA3 record: 'LightContext' object has no attribute 'get_resource'

### ISSUE THREAD

**Issue by joesavage** [ARM-software/workload-automation#813]
When trying to record a series of input events on the cutting edge of the `next` branch, I'm finding that things aren't quite working as expected.

Particularly, running `wa record -srt -w angrybirds_rio` as described in the docs [here](https://github.com/ARM-software/workload-automation/blob/next/doc/source/how_tos/users/revent.rst#recording) yields the following output:

```
$ wa record -srt -w angrybirds_rio
INFO     Setting up target
INFO     Deploying angrybirds_rio
ERROR      File "/work/workload-automation/wa/framework/entrypoint.py", line 120, in main
ERROR        sys.exit(command.execute(config, args))
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 127, in execute
ERROR        self.workload_record(args)
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 202, in workload_record
ERROR        workload.apk.initialize(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 662, in initialize
ERROR        self.resolve_package(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 678, in resolve_package
ERROR        self.resolve_package_from_host(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 716, in resolve_package_from_host
ERROR        apk_file = context.get_resource(ApkFile(self.owner,
ERROR    
ERROR    AttributeError('LightContext' object has no attribute 'get_resource')
```

Unlike `ExecutionContext`, the `LightContext` used in `wa record` doesn't seem to define `get_resource` and thus isn't playing nicely with the infrastructure in `workload.py`...

**Comment by setrofim** [comment]
Should be fixed by https://github.com/ARM-software/workload-automation/commit/4a06c1373884c27aa6724e33ca62b2fe5de87573

**Comment by joesavage** [comment]
That's giving me:

```
ERROR      File "/work/workload-automation/wa/framework/entrypoint.py", line 120, in main
ERROR        sys.exit(command.execute(config, args))
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 127, in execute
ERROR        self.workload_record(args)
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 202, in workload_record
ERROR        workload.apk.initialize(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 662, in initialize
ERROR        self.resolve_package(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 678, in resolve_package
ERROR        self.resolve_package_from_host(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 722, in resolve_package_from_host
ERROR        strict=self.strict)
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 295, in get_resource
ERROR        return self.resolver.get_resource(resource, strict)
ERROR    
ERROR    AttributeError('ResourceResolver' object has no attribute 'get_resource')
```

Looks like `get_resource` is implemented on `ExecutionContext` as a wrapper around `resolver.get` rather than on `ResourceResolver`.

**Comment by setrofim** [comment]
Argh, sorry -- that's my bad. Should be fixed new.

**Comment by joesavage** [comment]
That's that particular error solved. Looks like that wasn't the only issue with `LightContext` though:

```
$ wa record -srt -w angrybirds_rio
INFO     Setting up target
INFO     Deploying angrybirds_rio
ERROR      File "/work/workload-automation/wa/framework/entrypoint.py", line 120, in main
ERROR        sys.exit(command.execute(config, args))
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 127, in execute
ERROR        self.workload_record(args)
ERROR      File "/work/workload-automation/wa/commands/revent.py", line 203, in workload_record
ERROR        workload.apk.setup(context)
ERROR      File "/work/workload-automation/wa/framework/workload.py", line 665, in setup
ERROR        context.update_metadata('app_version', self.apk_info.version_name)
ERROR    
ERROR    AttributeError('LightContext' object has no attribute 'update_metadata')
```

I no longer need `wa record` for the moment so there's no pressure on my end for a quick fix, but thought it was worth keeping this tracked anyway.

**Comment by marcbonnici** [comment]
This should be fixed with 21b80798a32da74be666c4bf43dda00c691c663b

**Comment by joesavage** [comment]
Looks like that's solved it, thanks.
